src/server/rdb_save.cc

```diff
@@ -666,14 +666,16 @@ error_code RdbSerializer::SaveStreamConsumers(streamCG* cg) {
   return error_code{};
 }
 
-error_code RdbSerializer::SendFullSyncCut() {
+error_code RdbSerializer::SendFullSyncCut(uint64_t final_version) {
   RETURN_ON_ERR(WriteOpcode(RDB_OPCODE_FULLSYNC_END));
 
+  VLOG(1) << "final_version=" << final_version;
   // RDB_OPCODE_FULLSYNC_END followed by 8 bytes of 0.
   // The reason for this is that some opcodes require to have at least 8 bytes of data
   // in the read buffer when consuming the rdb data, and since RDB_OPCODE_FULLSYNC_END is one of
   // the last opcodes sent to replica, we respect this requirement by sending a blob of 8 bytes.
-  uint8_t buf[8] = {0};
+  uint8_t buf[sizeof(uint64_t)] = {0};
```

Comment:
no so sure I understand this line. You want to write uint64_t final_version
why do you do uint8_t buf[sizeof(uint64_t)] = {0};

Reply:
Hmm, I probably should fix it to write it through the endianity functions instead anyway. But what should I do instead?